packages/x-components/src/x-installer/x-installer/x-installer.ts

```diff
@@ -237,59 +239,51 @@ export class XInstaller {
    * of the plugin.
    * @param bus - The {@link @empathyco/x-bus#XBus} to be used to create the XPlugin.
    *
-   * @returns PluginObject<XPluginOption> - The plugin instance.
+   * @returns Plugin<XPluginOption> - The plugin instance.
    * @internal
    */
   protected installPlugin(
     pluginOptions: XPluginOptions,
     bus: XBus<XEventsTypes, WireMetadata>
-  ): PluginObject<XPluginOptions> {
+  ): Plugin<XPluginOptions> {
     const plugin = this.options.plugin ?? new XPlugin(bus);
-    const vue = this.getVue();
-    vue.use(plugin, pluginOptions);
+    this.getApp().use(plugin, pluginOptions);
     return plugin;
   }
 
   /**
-   * Install more plugins to Vue defined by the user.
+   * Runs the installExtraPlugins callback defined in the {@link InstallXOptions}
+   * to allow the user to install more plugins to the App.
    *
    * @param bus - The events bus used in the application.
-   * @returns The arguments from the plugins installation to be used in Vue's constructor.
+   * @returns An empty promise.
    * @internal
    */
-  protected installExtraPlugins(
-    bus: XBus<XEventsTypes, WireMetadata>
-  ): Promise<VueConstructorPartialArgument> {
-    const vue = this.getVue();
+  protected installExtraPlugins(bus: XBus<XEventsTypes, WireMetadata>): Promise<void> {
     return Promise.resolve(
-      this.options.installExtraPlugins?.({ vue, snippet: this.snippetConfig!, bus })
+      this.options.installExtraPlugins?.({ app: this.getApp(), snippet: this.snippetConfig!, bus })
     );
   }
 
   /**
-   * In the case that the `app` parameter is present in the {@link InstallXOptions}, then a new Vue
-   * application is created using that app.
+   * In the case that the `rootComponent` parameter is present in the {@link InstallXOptions}, then a new Vue
+   * application is created using that component as root.
    *
    * @param extraPlugins - Vue plugins initialisation data.
-   * @returns The Created Vue application or undefined if not created.
    *
    * @internal
    */
-  protected createApp(extraPlugins: VueConstructorPartialArgument): Vue | undefined {
-    if (this.options.app !== undefined) {
-      const vue = this.getVue();
-      const app = new vue({
+  protected createApp(extraPlugins: VueConstructorPartialArgument): void {
+    if (!this.options.app && this.options.rootComponent !== undefined) {
+      this.app = createApp(this.options.rootComponent, {
         ...extraPlugins,
         ...this.options.vueOptions,
         provide: {
-          snippetConfig: (this.snippetConfig = vue.observable(this.snippetConfig))
+          snippetConfig: this.snippetConfig
```

Comment:
We can no longer provide the snippet this way. We need to do it through app.provide:

```ts
      this.app = createApp(this.options.rootComponent, {});
      this.app.provide('snippetConfig', this.snippetConfig);
```